### ISSUE TITLE
syscall 0230 error on (arch)linux

### ISSUE THREAD

**Issue by poel22** [rstudio/rstudio#6379]
**problem:**
GUI gets white and just top menu bar stays but wont execute anything, at least it appears like that it doesnt do anything, i cant close the app gracefully from within the menu

if i start program from terminal, below dump is the result once the gui died

**when:**
randomly, but very often (every 10 minutes approx)
if i do basically anything (editing, running code, generating knitr document)
in any mode (project, r file, rmd file, ...)

**cause:**
i guess this is basically a (already fixed) chromium issue (glibc)
https://bugzilla.redhat.com/show_bug.cgi?id=1773289

**environment:**
arch linux with fully up to date programs & kernel (including rstudio) and high end hardware, i dont think thats the issue
r studio package installed: aur/rstudio-desktop-bin 1.2.5033-3

**dump:**
```
../../3rdparty/chromium/sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0230
Received signal 11 SEGV_MAPERR 0000000000e6
#0 0x7f3f3a55aa4f <unknown>
#1 0x7f3f3a55ae4b <unknown>
#2 0x7f3f3a55b4ce <unknown>
#3 0x7f3f3f716800 <unknown>
#4 0x7f3f3b5be863 <unknown>
#5 0x7f3f3b5c6e4a <unknown>
#6 0x7f3f3f716800 <unknown>
#7 0x7f3f367b12d1 __clock_nanosleep_2
#8 0x7f3f367b6bf7 __GI___nanosleep
#9 0x7f3f3a56770b <unknown>
#10 0x7f3f3a569618 <unknown>
#11 0x7f3f3b649f2d <unknown>
#12 0x7f3f3b649fd3 <unknown>
#13 0x7f3f3b6414a9 <unknown>
#14 0x7f3f3b6415fe <unknown>
#15 0x7f3f3b641677 <unknown>
#16 0x7f3f3b687300 <unknown>
#17 0x7f3f3c48d74f <unknown>
#18 0x7f3f3d380ea9 <unknown>
#19 0x7f3f39408ed8 <unknown>
#20 0x7f3f3940b8dc <unknown>
#21 0x7f3f3940be5f <unknown>
#22 0x7f3f3940c1b9 <unknown>
#23 0x7f3f39b81dee <unknown>
  r8: 00007fffc87b2bd0  r9: 00007fffc87b2bb0 r10: 00007fffc87b2b90 r11: 00007fffc87b2b70
 r12: 00007fffc87b2c80 r13: 00007fffc87b2c30 r14: 00000000012cda80 r15: 00007fffc87b2c80
  di: 0000000000000001  si: 00007fffc87b2a80  bp: 00007fffc87b2c60  bx: 00000000000000e6
  dx: 0000000000000000  ax: 0000000000000000  cx: 0000000000000012  sp: 00007fffc87b2c30
  ip: 00007f3f3b5be863 efl: 0000000000010202 cgf: 002b000000000033 erf: 0000000000000006
 trp: 000000000000000e msk: 0000000000000000 cr2: 00000000000000e6
[end of stack trace]
Calling _exit(1). Core file will not be generated.
```

**Comment by poel22** [comment]
downgrading glibc & lib32-glibc seems like a temp fix

**Comment by ronblum** [comment]
@poel22 Thank you for filing the issue and providing the workaround. I'll mark this as an issue to be triaged as we continue development of RStudio. Thanks!

**Comment by chendaniely** [comment]
I can confirm I am also having this issue in Arch Linux. This is the (same as poel22) output I get when I run rstudio in the terminal and it crashes:

```
$ rstudio-bin
../../3rdparty/chromium/sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0230
```

I am running the desktop-preview (v1.3.903) from the [AUR](https://aur.archlinux.org/packages/rstudio-desktop-preview-bin/) that I maintain.

**Comment by kevinushey** [comment]
Is there any hope that the offending change in `glibc` will be reverted?

**Comment by chendaniely** [comment]
Reading through the bugzilla comments from OP, it doesn't seem like it'll be reverted. The most relevent comment about the change is this

> Assuming 0230 is decimal, this corresponds to clock_nanosleep. Presumably, the existing seccomp filters in chromium only allow nanosleep, but not clock_nanosleep. They need to be adjusted.

Also, the Bugzilla comments link to the chromium bug report (which I guess is most relevant to rstudio): https://bugs.chromium.org/p/chromium/issues/detail?id=1025739

**Comment by arne-henningsen** [comment]
It seems that I experience the same bug that essentially makes RStudio unusable :-(((

`../../3rdparty/chromium/sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0230
Received signal 11 SEGV_MAPERR 0000000000e6
#0 0x7f372f3b2a4f <unknown>
#1 0x7f372f3b2e4b <unknown>
#2 0x7f372f3b34ce <unknown>
#3 0x7f373456f3c0 <unknown>
#4 0x7f3730416863 <unknown>
#5 0x7f373041ee4a <unknown>
#6 0x7f373456f3c0 <unknown>
#7 0x7f372b5f4341 clock_nanosleep
#8 0x7f372b5f9e97 nanosleep
#9 0x7f372f3bf70b <unknown>
#10 0x7f372ff963cb <unknown>
#11 0x7f372ff91bb7 <unknown>
#12 0x7f372ff91d38 <unknown>
#13 0x7f372ffca018 <unknown>
#14 0x7f3731906222 <unknown>
#15 0x7f37319063ca <unknown>
#16 0x7f373190640b <unknown>
#17 0x7f372f38c4d3 <unknown>
#18 0x7f372f3bfa10 <unknown>
#19 0x7f3734563609 start_thread
#20 0x7f372b636153 clone
  r8: 00007f3720ebd010  r9: 00007f3720ebcff0 r10: 00007f3720ebcfd0 r11: 00007f3720ebcfb0
 r12: 00007f3720ebd0c0 r13: 00007f3720ebd070 r14: 0000000002392210 r15: 00007f3720ebd0c0
  di: 0000000000000001  si: 00007f3720ebcec0  bp: 00007f3720ebd0a0  bx: 00000000000000e6
  dx: 0000000000000000  ax: 0000000000000000  cx: 0000000000000012  sp: 00007f3720ebd070
  ip: 00007f3730416863 efl: 0000000000010202 cgf: 002b000000000033 erf: 0000000000000006
 trp: 000000000000000e msk: 0000000000000000 cr2: 00000000000000e6
[end of stack trace]
Calling _exit(1). Core file will not be generated.`

**Comment by arne-henningsen** [comment]
This bug still occurs in RStudio preview version 1.3.921:
```
../../3rdparty/chromium/sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0230
```

**Comment by arne-henningsen** [comment]
The easiest ways (for me) to reproduce this error are (a) moving the vertical bar that separates the two panes on the left from the two panes on the right to the left or right and (b) opening an Sweave/knitr/Rnw file, clicking at "Compile PDF", and waiting for the PDF file to open (which crashes RStudio).

**Comment by skinamarink** [comment]
Getting the exact same error, with the same dump as shown above. Arch linux, version 1.2.5033-3. Crashes every 10 minutes, but can be reliably reproduced by:
1. Clicking "pop out window" on the code editor
2. Maximize RStudio window
3. Click on title bar, move window around

RStudio desktop is basically unusable now...

**Comment by kevinushey** [comment]
Sorry, but this isn't going to be fixed for RStudio v1.3.

**Comment by arne-henningsen** [comment]
If this bug won't be fixed soon, RStudio won't be usable on any recent Linux system soon!!!  :-(((

Can anyone recommend a good alternative to RStudio?

**Comment by gtritchie** [comment]
This has been backported to Qt 5.14: https://bugreports.qt.io/browse/QTBUG-81313

**Comment by chendaniely** [comment]
@arne-henningsen I've been using VS Code with rstudio for a while now,. That may be your best bet to get some R IDE working. You can install the R extensions for VS Code:

- https://marketplace.visualstudio.com/items?itemName=Ikuyadeu.r
- https://marketplace.visualstudio.com/items?itemName=REditorSupport.r-lsp

This post by Miles Mcbain may help as well: https://milesmcbain.xyz/alt-r-with-vscode/

And then there's emacs + ess :)

**Comment by gtritchie** [comment]
Marking as 1.3, but we are blocked until Qt 5.12.8 is released, hopefully by end of March.

**Comment by ssoftcheck** [comment]
maybe a temporary solution is to run rstudio server locally and use through the browser?

**Comment by edsandorf** [comment]
@arne-henningsen I encountered the very same issue. It's been bugging me for days. I have been experimenting with Atom (since I use it for all non-R stuff) using the Hydrogen package. It's possible to set up for interactive use, but it is quite cumbersome and not a long-term solution. You can find many resources for how to set it up by Googling "Atom and R". 

Here is my error output

```
../../3rdparty/chromium/sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0230
Received signal 11 SEGV_MAPERR 0000000000e6
#0 0x7fb8e0a05a4f <unknown>
#1 0x7fb8e0a05e4b <unknown>
#2 0x7fb8e0a064ce <unknown>
#3 0x7fb8e5bc1800 <unknown>
#4 0x7fb8e1a69863 <unknown>
#5 0x7fb8e1a71e4a <unknown>
#6 0x7fb8e5bc1800 <unknown>
#7 0x7fb8dcc5c2d1 __clock_nanosleep_2
#8 0x7fb8dcc61bf7 __GI___nanosleep
#9 0x7fb8e0a1270b <unknown>
#10 0x7fb8e15e93cb <unknown>
#11 0x7fb8e15e4bb7 <unknown>
#12 0x7fb8e15e4d38 <unknown>
#13 0x7fb8e161d018 <unknown>
#14 0x7fb8e2f59222 <unknown>
#15 0x7fb8e2f593ca <unknown>
#16 0x7fb8e2f5940b <unknown>
#17 0x7fb8e09df4d3 <unknown>
#18 0x7fb8e0a12a10 <unknown>
#19 0x7fb8e5bb646f start_thread
#20 0x7fb8dcc943d3 __GI___clone
  r8: 00007fb8b7ffe010  r9: 00007fb8b7ffdff0 r10: 00007fb8b7ffdfd0 r11: 00007fb8b7ffdfb0
 r12: 00007fb8b7ffe0c0 r13: 00007fb8b7ffe070 r14: 0000000002238a20 r15: 00007fb8b7ffe0c0
  di: 0000000000000001  si: 00007fb8b7ffdec0  bp: 00007fb8b7ffe0a0  bx: 00000000000000e6
  dx: 0000000000000000  ax: 0000000000000000  cx: 0000000000000012  sp: 00007fb8b7ffe070
  ip: 00007fb8e1a69863 efl: 0000000000010202 cgf: 002b000000000033 erf: 0000000000000006
 trp: 000000000000000e msk: 0000000000000000 cr2: 00000000000000e6
[end of stack trace]
Calling _exit(1). Core file will not be generated.
```

**Comment by poel22** [comment]
as i already mentioned, downgrading and adding glibc and lib32-glibc to IgnorePkg fixes the issue and for me everything else still works fine. i know its not the "arch" kind of way but i am pragmatic there.

**Comment by arne-henningsen** [comment]
@chendaniely @ssoftcheck @edsandorf @poel22 Thanks a lot for your recommendations! I find downgrading glibc on my very well working Ubuntu system too risky ("never change a running system" ...). As I am very much used to the RStudio keyboard shortcuts and I hope that I can use RStudio Desktop soon again, I installed RStudio Server locally (which took me less than 5 minutes) and I am now using it through a web browser (which works very well :-) )

**Comment by jmcphers** [comment]
We are not updating Qt right before the first 1.3 release; this will probably need to happen in a followup 1.3 release.

**Comment by chendaniely** [comment]
@poel22 which version of glibc and lib32-glibc did you downgrade to? 2.31-1 or 2.30-3? or something earlier?

**Comment by poel22** [comment]
> @poel22 which version of glibc and lib32-glibc did you downgrade to? 2.31-1 or 2.30-3? or something earlier?
sorry for not responding, i downgraded to 2.30-3

**Comment by gtritchie** [comment]
Still waiting on Qt 5.12.8 to be released. Was supposed to be last Friday, they haven't updated their dates yet. https://wiki.qt.io/Qt_5.12_Release

**Comment by gtritchie** [comment]
Qt 5.12.8 now released and has the fix:
>  - [QTBUG-81313] Enabled running with glibc 2.30 in Linux sandbox

**Comment by gtritchie** [comment]
This is not yet closed for 1.3, but I've merged the fix for master (1.4) via https://github.com/rstudio/rstudio/pull/6624

**Comment by gtritchie** [comment]
Fixed in 1.3-patch via 8a747fec

**Comment by chendaniely** [comment]
thanks for fixing the issue!

for everyone else in the thread i just created a daily AUR package if you still want to run the desktop version. I need to figure out how to keep it upto date without me checking everyday, but I'll be using it for the time being.

**Comment by arne-henningsen** [comment]
The bug still occurs in RStudio Preview Release v1.3.956-2 (released on April 30, 2020). I hope that a package for Ubuntu that includes the bug fix will soon be available.

**Comment by samuelkn** [comment]
I encountered the same issue on manjaro. Upgraded to the rstudio-desktop-daily-bin (1.4.314) version on AUR yesterday, and everything works fine until now. Qt version is 5.14.2 and glibc 2.31.

**Comment by edsandorf** [comment]
I tried the 1.4.314 version from AUR and it still doesn't work for me. It appears to crash when I go full-screen (3840x2160) and move the panes around. 

Error message is still: 
```
../../3rdparty/chromium/sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0230
```

```
QMake version 3.1
Using Qt version 5.14.2 in /usr/lib
```

```
ldd (GNU libc) 2.31
```

Looks like I am back to using rstudio-server through the browser for now.

**Comment by dr-emacs** [comment]
Did you try the workaround suggested in #6782 ?

**Comment by gtritchie** [comment]
There is a problem with the 1.4 builds, so am reactivating this while I work on the fix. Also tracked in #6782 but will leave both open due to number of conversations in each!

**Comment by gtritchie** [comment]
This is now fixed starting with daily build 1.4.325. The fix will also be in the first bug-fix release for 1.3 (known as 1.3-patch) but no builds available for that yet.

We will are not currently planning to fix this for RStudio 1.2 (any version).

**Comment by SimonCoulombe** [comment]
I had the same error on rstudio 1.2, so I upgraded to the latest daily build.  

I can't run rstudio 1.4.358, because it requires libssl1.0.0 and I am on ubuntu 20.04

Any way to get rstudio working on ubuntu 20.04?

**Comment by arne-henningsen** [comment]
@SimonCoulombe: I use RStudio 1.3.956 from https://rstudio.com/products/rstudio/download/preview/ on my Ubuntu 20.04 system and start RStudio with:
export RSTUDIO_CHROMIUM_ARGUMENTS="--disable-seccomp-filter-sandbox"; rstudio
as suggested in #6782 as a workaround. This works well :-)

**Comment by SimonCoulombe** [comment]
it works!
thanks for your help

**Comment by seanpor** [comment]
With the 1.3.959 release, I get a blank screen on popop 20.04 if I maximize.  Using Kevin's debugging page, I narrowed it down.
I get the message:
 ../../3rdparty/chromium/sandbox/linux/seccomp-bpf-helpers/sigsys_handlers.cc:**CRASHING**:seccomp-bpf failure in syscall 0230
using Options / General / Advanced / OS Integration / Rendering engine = 'software' is the only fix I found.

**Comment by gtritchie** [comment]
Fix for this is coming in the first bug-fix release of 1.3, now available as a preview: https://rstudio.com/products/rstudio/download/preview/

**Comment by R-Hannibal** [comment]
Hey @poel22, thanks again for finding and reporting this issue as well as the workaround fix you discovered. We now have a code fix in place in our latest preview build (available [HERE](https://rstudio.com/products/rstudio/download/preview/)).

If you get the chance will you let us know if the issue is resolved for you in that build?

**Comment by samuelkn** [comment]
I'm running the 1.3.1004 version since yesterday. Everything went fine so far...

**Comment by SimonCoulombe** [comment]
working for me as well since yesterday, kubuntu 20.04

**Comment by rjm001** [comment]
Hi Everyone. Thanks for contributing to this thread. I have been following it and experiencing the same issues for the last two months. I am running popos 20.04. I was nervous to use the "downgrading-glibc" workaround for the same reasons discussed by arne-henningsen. The other, "export RSTUDIO_CHROMIUM_ARGUMENTS...", workaround would buy me longer sessions but my Rstudio sessions were still eventually crashing and returning the errors as described above. Hence, for the last few months I have been using just a text editor and R from the terminal to do data analysis.

Happy to note that I am no longer experiencing this issue with the 1.3.1008 preview edition. (Note, as of June 7th, the default build Rstudio directs you to--at https://rstudio.com/products/rstudio/download/#download--version 1.3.959, still has the issue, but the preview build--at https://rstudio.com/products/rstudio/download/preview/--has the issue solved. Also note that, (I am guessing because of the way they are numbered), Eddy thought 1.3.1008 was a "downgrade" from 1.3.959 after I downloaded it. So tell Eddy to "downgrade" your Rstudio version knowing that it really is an upgrade. 

Thanks very much to the team that solved this!

**Comment by SaraTohidi** [comment]
I have the same issue. Since I install [RStudio 1.3.1004 version,](https://rstudio.com/products/rstudio/download/preview/) everything works fine.
Thanks.

**Comment by R-Hannibal** [comment]
Thanks for confirming the fix to all who commented!

**Comment by SimonCoulombe** [comment]
Is this only an issue on RStudio desktop?

I am having random crashes on RStudio server and I was wondering if this could be the same issue. I 'm not sure where to look for the logs as I am only a user on that server.

thanks

**Comment by gtritchie** [comment]
This issue was RStudio Desktop specific.

**Comment by R-Hannibal** [comment]
@SimonCoulombe - thanks so much for contributing to this issue thread and helping us track down the bug.

As @gtritchie mentioned the issue we addressed through this ticket was desktop specific, so if you're still experiencing crash issues on RStudio server we should probably open a new ticket.

Logs would be very helpful in that new ticket to help us track down the issue, so please consult the [Automated Crash Reporting](https://docs.rstudio.com/ide/server-pro/latest/server-management.html#automated-crash-reporting) section of our admin guide for how to capture crash information.

**Comment by SimonCoulombe** [comment]
thank YOU for fixing it!   -- I'll set that up .
